### ISSUE TITLE
[Q] Sweep job finished before running the whole code

### ISSUE THREAD

**Issue by 3i-MiaHong** [wandb/wandb#5607]
I create a Sweep by below code.
`wandb.agent(sweep_id, function=train, count=2)` 

and I define sweep_id before above line, and also train function.

when I run this python file, the first Job runs well.
But from the second one, it finished before running the whole code with below message.
```log
wandb: Waiting for W&B process to finish... (failed 1). Press Control-C to abort syncing.
```   

so I'm using NanoDet, and this is part of my train function.
```python
def train(config=None):
    run = wandb.init(config=config, reinit=True)
    config = wandb.config

    rotation = config.rotation
    shear = config.shear
    translate = config.translate
    brightness = config.brightness

    # between here, all NanoDet-Plus code

    trainer.fit(task, train_dataloader, val_dataloader,
                ckpt_path=model_resume_path)

    run.finish()
```

First one runs well, and from the second one, the sender send exit.
```log
DEBUG   SenderThread:186 [sender.py:send():375] send: telemetry
DEBUG   SenderThread:186 [sender.py:send():375] send: exit
INFO    SenderThread:186 [sender.py:send_exit():598] handling exit code: 1
INFO    SenderThread:186 [sender.py:send_exit():600] handling runtime: 0
INFO    SenderThread:186 [sender.py:_save_file():1378] saving file wandb-summary.json with policy end
INFO    SenderThread:186 [sender.py:send_exit():606] send defer
```

Is there anybody who can help to handle this?

**Comment by nate-wandb** [comment]
Hi @3i-MiaHong, is there a reason you are setting `reinit=True`? I'm not sure that this is the cause but you should be able to remove this since the agent can handle running multiple `wandb.init`'s. Could you try this and let me know if you still see this issue?

**Comment by 3i-MiaHong** [comment]
Hi @nate-wandb ! thank you for replying.
Actually at first I ran the code without `reinit=True`, and it wasn't going well, so I add that by referring someone's code.

**Comment by nate-wandb** [comment]
Hi @3i-MiaHong, sorry for the delay. A couple questions that may help:
1. What version of `wandb` are you using? 
2. Are you able to share the debug logs for both runs? 
3. Could you also share how you initiate the sweep as well?

Thank you,
Nate

**Comment by nate-wandb** [comment]
Hi @3i-MiaHong, I wanted to follow up and see if this was still an issue for you?

**Comment by 3i-MiaHong** [comment]
hi @nate-wandb, sorry for the delay.

Actually I solved the issue, and it was not a W&B issue.

I think W&B Sweeps set random hyper parameter value and starts first run. After that, set second value and starts second run, and compare the second value with the model's default config's value. and maybe it's different, training will be start.

I'm not sure that is correct what I understood, but at least in my case, this was the reason. Also I found an error log about the type error.
So I modify model config value 0 to 0.0, and then the issue was solved.
